### ISSUE TITLE
AO 2.10 Changelogs Round 2

### ISSUE THREAD

**Issue by Crystal2002** [Crystal2002/AO-Documentation#4]
Okay, that's it. I'm re-writing all this, and just categorizing **EVERY. CHANGE.** made to the client (that an end-user _might_ care about?). Just because paging through 7 pages of commits is ass for people. Also because I don't want to ping Salanto's email to hell and back for the next couple of hours, because _I have no life tonight_.

[Last left off at Pull Request `#658`]

**NOTE: I HAVE NOT LOOKED ANY DEEPER INTO THE PRs OTHER THAN THE NAMES, OR _MAYBE_ THE FIRST LINE WITHIN IT. TAKE AT YOUR OWN RISK.**

**Major Changes:**
- Music List Streaming over ASS Packet
- Add Base Folder Mounting
- The following Client-Side Commands have been Removed:
  - `/pair` (Part of an earlier PR)
  - `/settings`
  - `/offset`
  - `/voffset`
  - `/switch_am`
  - `/enable_blocks`
  - `/non_int_pre`
  - `/save_chatlog`
  - `/load_case`
- Added compatibility with tracker formats (i.e. MIDI)
  - (Someone get sD to write up how the fuck this works. Please.)
- Switched to Masterserver 3!!!!
- Masterserver Chat is no more. Long Reign **_Debug Logs_**, which will assist the tech-savvy (or the developers) in solving tech issues!
- Character Select Listing now updates when the server updates it's CharSelect files. No more rejoining required.
- Icons now appear in Position, INISwap, and Emote Dropdowns (just like with the Effects Dropdown).
  - Position just miniaturizes it's BG
  - INIswap uses `char_icon`
  - Emote uses the Emote Buttons
- 

**Minor Changes:**
- Accounting for pathing differences between theme & misc when getting data for Effects
- Remember Search Terms when swapping between Area/Music List
- Audio Device is reinitialized when the Default Audio Device is lost/disconnected
- Improved Viewport performance by preloading next frame before the current frame is complete
- Users can now configure the log's timestamp format via Settings Menu
- Client Version Number is now visible in the Window Title
- When a Notice Window appears, the `OK` button will be disabled with a countdown next to it, instead of remaining hidden until the countdown was over.

**Bugfixes:** (Don't know what's a 2.10-dev Bugfix vs a 2.9.1 Bugfix, so someone smarter than me can handle that)
- Fix 1st Server in Serverlist not initiating connections unless another server was selected first
- Fix Custom Shouts not playing Audio
- Fixes several bugs with Pairing
  - `Person #2` would disappear whenever `Person #1` talked, and vice-versa.
  - When paired, the idle animation would play only once.
- Fixes file lookup to check if a streamed asset is available to be used, if not found locally.
- Fixes BG Positions w/o a desk inheriting the previous position's desk.
- Fixes `(c)` animations not resolving into the `(a)` animation
  - For those who may have missed it in 2.9, `(c)` are optional animations that are used for transitioning from `(b)` [Talking] to `(a)` [Idle] animations.
- Fix non-image files from being loaded as images
- Fix Custom Shouts appearing on characters without the option
- Fix an animation freeze when a music file is being loaded
  - Was most noticeable when streaming or loading large music files.

**[PRs I skipped over, either because it's tech-y or I have no idea WTF is happening here]**
- Swapping "UNUSED" macro for "Q_UNUSED"
- Handling Extending Desk Modifiers on Chat Tick & not on Preanim Complete
- Fix excessive calls to ARUP_append
- Fully Asynchronous Anim. Loading
- Extend /pair to accept CharNames & IDs
  - Removed? I'm assuming this was related to the clientside command at least?
- Fixed Linux Compilation Error due to outdated Qt Versions
- Sanity Check before Removing [SFX Dropdown?]
- Remove Client-Side Waiting
- Fix emote buttons not generating due to bad VPath Lookup
  - Was this when mounted emotes weren't appearing?
- Don't restart background if background is unchanged
- Fix BGs getting messed up
  - I think this was when changing the theme would screw around with how the background was showing, and it would never reposition itself properly? Was a 2.10-dev issue, I think, so who caaaaares?
- Fix Segmentation Fault w/ Music
   - No idea what was going on, was this with the race condition bullshit? Anyone mind explaining in baby terms what was happening?

**Comment by stonedDiscord** [comment]
the tracker format files work like normal music files.

the midi files can have a soundfont which needs to be the same filename but instead of .mid it's .sf2

**Comment by Crystal2002** [comment]
Next batch, because lazy to just edit the original post. Might do that tomorrow when I'm not ready to fall over dead 

[Stopped @ 712]

Major:
* Allows for setting the Blip Rate to 0 (means the blip only goes off once per message) (659)
* Massive Effects System overhaul! (663) (701)
* The client can now connect to servers via TCP & Websockets! (696)
  * (Can a Networking Wiz can baby this into basic terms plz)

Minor:
* Add showname & selfname colors to `courtroom_fonts.ini` (672)
  * Allows for the name attached to _your_ messages to have a different color from the rest!
* Add ability to change the timestamps' colors in `courtroom_fonts.ini` (698)

Bugfix:
* Fixes the client's heartbeat from triggering constantly
* Shout no longer force preanimations from triggering (697)
* Fixes emote buttons from sometimes scaling with Nearest Neighbor (699)
* Fixes the soundlist from treating the Alias as the actual Audiofile Name (708)
* Fixes iniswap_remove button perma-removing the entry from `iniswaps.ini`. (710)
* Fixes issue where iniswap dropdown was destructive to `char.ini` (712)
  * Note that for the time being, your INIswap will NOT be remembered. This is a side-effect of fixing the error.

Unknown:
* Improved Evidence End-User Experience (660)
  * Never used Evidence, so uh. What happened, again?
* Fix Pos Dropdown failing to properly populate (661)
* Fix demoserver max_wait skips (680)
* Another race condition thingy (700)
* AOImage Masking (705)
  * I know what the Issue being fixed was, but I'm confused w/ the language as to if this is now a True/False someplace in the theme's files or if it's just reverting it to original behavior altogether

**Comment by Salanto** [comment]
The client can now connect to servers via TCP & Websockets! (696)
(Can a Networking Wiz can baby this into basic terms plz)

TLDR : The client now supports both the protocol WebAO uses and the legacy one. Only really relevant when you run Cloudflare tunnels or some other Websocket DDOS protection.

**Comment by Salanto** [comment]
Fixes issue where iniswap dropdown was destructive to char.ini (712)
Note that for the time being, your INIswap will NOT be remembered. This is a side-effect of fixing the error.

Good. An iniswap should not be remembered, period.

**Comment by Salanto** [comment]
If its race condition and has anything to do with music, don't mention it and just put "Asynch music loading" on the label.
